src/kern/exec.c

```diff
@@ -17,11 +17,42 @@
 #include <bpf/bpf_helpers.h>
 #include <bpf/bpf_tracing.h>
 
+/*
+ * Define a map structure to communicate with the userspace program.
+ * The type of map is BPF_MAP_TYPE_PERCPU_ARRAY;
+ * it contains one element, found for key = 0
+ * and the value is an array with number_of_CPUs elements.
+ */
+struct bpf_map_def SEC("maps") output_map = {
+        .type = BPF_MAP_TYPE_PERCPU_ARRAY,
```

Comment:
I see that you are using 8 spaces for indentation, but the Linux kernel style actually requires "tabulation" characters. They recommend that you configure your editor to show tabs as 8 spaces because there are times when you want to align something using spaces (eg if your function arguments span two lines, the second line should be aligned with the first one using tabs and then spaces) but your code should actually not contain spaces.

I went back to look at the style guide and I reckon it is really stupidly worded: "Outside of comments, documentation and except in Kconfig, spaces are never used for indentation, and the above example is deliberately broken." so basically they give you an example but you shouldn't follow it, aerrm.

If you look at a random function, say https://github.com/torvalds/linux/blob/master/kernel/torture.c#L204 you can see that indentation is done with one character (tab)

Anyway, let's fix this separately, this is one of those things that deserve a separate commit (or PR) to keep each commit "atomic" as I explained above.